core-bundle/src/Filesystem/Dbafs/Dbafs.php

```diff
@@ -185,18 +185,19 @@ public function setExtraMetadata(string $path, array $metadata): void
         $event = new StoreDbafsMetadataEvent(
             $this->table,
             $row,
-            // Remove non-matching columns before dispatching event
-            array_intersect_key($metadata, $columnFilter)
+            $metadata,
         );
 
         $this->eventDispatcher->dispatch($event);
 
-        $this->connection->update(
-            $this->table,
-            // Filter columns again before performing the query
-            array_intersect_key($event->getRow(), $columnFilter),
-            ['uuid' => $uuid]
-        );
+        if ($data = array_intersect_key($event->getRow(), $columnFilter)) {
```

Comment:
if the event returns zero fields, we must not update the database

Reply:
True, I hate this behavior of DBAL…